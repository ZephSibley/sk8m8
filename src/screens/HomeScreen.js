import React from 'react';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import Typography from '@material-ui/core/Typography';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import axios from 'axios';

import '../styles/screens.css'
import UserCard from '../components/account/UserCard';
import UpdateEmailForm from '../components/ui/forms/UpdateEmailForm';
import UpdatePasswordForm from '../components/ui/forms/UpdatePasswordForm';

// CHANGE ME IN PRODUCTION
const userDetailsMock = {username: 'zeph', avatar:'', status:'bio bio bio'}


const HomeScreen = () => {

    const [expanded, setExpanded] = React.useState(false);

    const togglePanel = panel => (event, isExpanded) => {
        setExpanded(isExpanded ? panel : false);
    };

    return (
        <div className={'screen'}>

            <UserCard
                requests={axios}
                username={userDetailsMock.username}
                avatar={userDetailsMock.avatar}
                bio={userDetailsMock.bio}
            />

            <ExpansionPanel
                expanded={expanded === 'emailPanel'}
                onChange={togglePanel('emailPanel')}
            >
                <ExpansionPanelSummary
                    expandIcon={<ExpandMoreIcon />}
                    aria-controls="update-email"
                    id="update-email-header"
                >
                    <Typography>Update Email</Typography>
                </ExpansionPanelSummary>
                <ExpansionPanelDetails>
                    <UpdateEmailForm />
                </ExpansionPanelDetails>
            </ExpansionPanel>
            <ExpansionPanel
                expanded={expanded === 'passwordPanel'}
                onChange={togglePanel('passwordPanel')}
            >
                <ExpansionPanelSummary
                    expandIcon={<ExpandMoreIcon />}
                    aria-controls="update-password"
                    id="update-password-header"
                >
                    <Typography>Update Password</Typography>
                </ExpansionPanelSummary>
                <ExpansionPanelDetails>
                    <UpdatePasswordForm />
                </ExpansionPanelDetails>
            </ExpansionPanel>
        </div>
    );
}

export default HomeScreen;