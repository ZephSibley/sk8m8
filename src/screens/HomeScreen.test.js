import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from './HomeScreen';

test('Basic Rendering', () => {
    const component = renderer.create(
        <HomeScreen />
    )
    let tree = component.toJSON
    expect(tree).toMatchSnapshot();
})