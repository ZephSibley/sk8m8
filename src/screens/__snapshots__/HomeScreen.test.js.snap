// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering 1`] = `
<div
  className="screen"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-avatar"
      >
        <div
          className="MuiAvatar-root"
        >
          <img
            alt="Loading"
            className="MuiAvatar-img"
            src="ajax-loader.gif"
          />
        </div>
      </div>
      <div
        className="MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
        >
          Loading
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body2"
      >
        
      </p>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-rounded"
  >
    <div
      aria-controls="update-email"
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root"
      id="update-email-header"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Update Email
        </p>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="update-email-header"
            id="update-email"
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root"
            >
              <form
                className="makeStyles-form-106"
                onSubmit={[Function]}
              >
                <div
                  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink={false}
                  >
                    email
                  </label>
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input MuiInput-input"
                      disabled={false}
                      name="email"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="email"
                      value=""
                    />
                  </div>
                  
                </div>
                <div
                  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink={false}
                  >
                    confirm email
                  </label>
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input MuiInput-input"
                      disabled={false}
                      name="confirm_email"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="email"
                      value=""
                    />
                  </div>
                  
                </div>
                <button
                  className="MuiButtonBase-root MuiButton-root makeStyles-submit-108 MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
                  disabled={true}
                  margin="normal"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  type="submit"
                >
                  <span
                    className="MuiButton-label"
                  >
                    Update Email
                  </span>
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-rounded"
  >
    <div
      aria-controls="update-password"
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root"
      id="update-password-header"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Update Password
        </p>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={-1}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="update-password-header"
            id="update-password"
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root"
            >
              <form
                className="makeStyles-form-106"
                onSubmit={[Function]}
              >
                <div
                  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink={false}
                  >
                    password
                  </label>
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input MuiInput-input"
                      disabled={false}
                      name="password"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="password"
                      value=""
                    />
                  </div>
                  
                </div>
                <div
                  className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink={false}
                  >
                    confirm password
                  </label>
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input MuiInput-input"
                      disabled={false}
                      name="confirm_password"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="password"
                      value=""
                    />
                  </div>
                  
                </div>
                <button
                  className="MuiButtonBase-root MuiButton-root makeStyles-submit-108 MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
                  disabled={true}
                  margin="normal"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  type="submit"
                >
                  <span
                    className="MuiButton-label"
                  >
                    Update Password
                  </span>
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
